docs/reference/modules/cluster/shards_allocation.asciidoc

```diff
@@ -114,28 +134,46 @@ of shards, and then moving shards between nodes to reduce the weight of the
 heavier nodes and increase the weight of the lighter ones. The cluster is
 balanced when there is no possible shard movement that can bring the weight of
 any node closer to the weight of any other node by more than a configurable
-threshold. The following settings allow you to control the details of these
-calculations.
+threshold. The weight of a node depends on the number of shards it holds and on
+the resource usage of each shard expressed in terms of the size of the shard on
+disk and the number of threads needed to support write traffic to the shard. The
+following settings control how {es} combines these values into an overall
+measure of each node's weight.
 
 `cluster.routing.allocation.balance.shard`::
-     (<<dynamic-cluster-setting,Dynamic>>)
-     Defines the weight factor for the total number of shards allocated on a node
-     (float). Defaults to `0.45f`. Raising this raises the tendency to
-     equalize the number of shards across all nodes in the cluster.
+(float, <<dynamic-cluster-setting,Dynamic>>)
+Defines the weight factor for the total number of shards allocated to each node.
+Defaults to `0.45f`. Raising this value increases the tendency of {es} to
+equalize the total number of shards across nodes ahead of the other balancing
+variables.
 
 `cluster.routing.allocation.balance.index`::
-     (<<dynamic-cluster-setting,Dynamic>>)
-     Defines the weight factor for the number of shards per index allocated
-      on a specific node (float). Defaults to `0.55f`. Raising this raises the
-      tendency to equalize the number of shards per index across all nodes in
-      the cluster.
+(float, <<dynamic-cluster-setting,Dynamic>>)
+Defines the weight factor for the number of shards per index allocated to each
+node. Defaults to `0.55f`. Raising this value increases the tendency of {es} to
+equalize the number of shards of each index across nodes ahead of the other
+balancing variables.
+
+`cluster.routing.allocation.balance.disk_usage`::
+(float, <<dynamic-cluster-setting,Dynamic>>)
+Defines the weight factor for balancing shards according to their predicted disk
+size in bytes. Defaults to `2e-11f`. Raising this value increases the tendency
+of {es} to equalize the total disk usage across nodes ahead of the other
+balancing variables.
+
+`cluster.routing.allocation.balance.write_load`::
+(float, <<dynamic-cluster-setting,Dynamic>>)
+Defines the weight factor for the write load of each shard, in terms of the
```

Comment:
Is this the place where we should mention that only the write-index of data streams have a write-load?

Reply:
I tried but didn't really like it. I added a paragraph further up about how resource usage is estimated.